### DISCUSSION TITLE
Removing header from the Manifest

### DISCUSSION THREAD

**Discussion by alshamams** [eclipse-pde/eclipse.pde#939] [Q&A]
I was trying to implement https://github.com/eclipse-pde/eclipse.pde/issues/864 and want to remove the existing header `Eclipse-BundleShape` in response to selection of default shape mode. This is what I am using:

```java
Map<String, IManifestHeader> headers = ((Bundle) bundle).getHeaders();
Object entry = headers.remove(ICoreConstants.ECLIPSE_BUNDLE_SHAPE);
IBundleModel model = bundle.getModel();
model.fireModelChanged( new ModelChangedEvent(model, IModelChangedEvent.REMOVE, new Object[] { entry }, null));
```

For some reason, this doesn't seem to work. I can confirm that the header is indeed removed from the model by printing the current list of headers but the UI is not reflecting that - the Manifest Editor continues to display the removed header. 
Any ideas?

**Comment by laeubi** [comment]
@alshamams sadly this is quite cumbersome to manipulate the model, the best is to look how the manifest editor do it e.g. if you change the Bundle name or version.

**Comment by gireeshpunathil** [comment]
somewhat related: https://bugs.eclipse.org/bugs/show_bug.cgi?id=411552 but still doesn't have a solution. Apart from the above recommendation, I suggest you debug `model.fireModelChanged` and see if it successfully completes or falls back due to some `conditions` and see what is missing.
